cmd/mt-parrot/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"github.com/grafana/metrictank/cmd/mt-fakemetrics/out"
+	"github.com/grafana/metrictank/cmd/mt-fakemetrics/out/gnet"
+	"github.com/grafana/metrictank/logger"
+	"github.com/grafana/metrictank/schema"
+	"github.com/raintank/met/statsd"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"time"
+)
+
+var (
+	gatewayAddress        string
+	gatewayKey            string
+	orgId                 int
+	partitionCount        int32
+	partitionMethodString string
+	testMetricsInterval   time.Duration
+	queryInterval         time.Duration
+	lookbackPeriod        time.Duration
+	logLevel              string
+	lastPublish           int64
+
+	partitionMethod schema.PartitionByMethod
+	publisher       out.Out
+)
+
+func init() {
+	parrotCmd.Flags().StringVar(&gatewayAddress, "gateway-address", "http://localhost:6059", "the url of the metrics gateway")
+	parrotCmd.Flags().StringVar(&gatewayKey, "gateway-key", "", "the bearer token to include with gateway requests")
+	parrotCmd.Flags().IntVar(&orgId, "org-id", 1, "org id to publish parrot metrics to")
+	parrotCmd.Flags().Int32Var(&partitionCount, "partition-count", 8, "number of kafka partitions in use")
+	parrotCmd.Flags().StringVar(&partitionMethodString, "partition-method", "bySeries", "the partition method in use on the gateway, must be one of bySeries|bySeriesWithTags|bySeriesWithTagsFnv")
+	parrotCmd.Flags().DurationVar(&testMetricsInterval, "test-metrics-interval", 10*time.Second, "interval to send test metrics")
+	parrotCmd.Flags().DurationVar(&queryInterval, "query-interval", 10*time.Second, "interval to query to validate metrics")
+	parrotCmd.Flags().DurationVar(&lookbackPeriod, "lookback-period", 5*time.Minute, "how far to look back when validating metrics")
+
+	parrotCmd.Flags().StringVar(&logLevel, "log-level", "info", "log level. panic|fatal|error|warning|info|debug")
+
+	formatter := &logger.TextFormatter{}
+	formatter.TimestampFormat = "2006-01-02 15:04:05.000"
+	log.SetFormatter(formatter)
+}
+
+func main() {
+	err := parrotCmd.Execute()
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+var parrotCmd = &cobra.Command{
+	Use:   "mt-parrot",
+	Short: "generate deterministic metrics for each metrictank partition",
+	Run: func(cmd *cobra.Command, args []string) {
+		lvl, err := log.ParseLevel(logLevel)
+		if err != nil {
+			log.Fatalf("failed to parse log-level, %s", err.Error())
+		}
+		if int(lookbackPeriod.Seconds())%int(testMetricsInterval.Seconds()) != 0 {
```

Comment:
this doesn't work accurately due to rounding.
also i don't think we even need this condition.

Reply:
What's the problem here?
```
./mt-parrot --lookback-period=10s --test-metrics-interval=4s
2020-03-17 10:01:01.168 [FATAL] lookback period must be evenly divisible by test metrics interval
```